src/common/outputService/deployOutputService.ts

```diff
@@ -0,0 +1,261 @@
+        break;
+
+      case AsyncOperationType.VERSIONED_PROMOTE:
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        summary = await this.processVersionedDeploy(branch, queryResp.sf_devops__Change_Bundle_Installs__r!.records);
+        this.printVersionedOrSoupDeploySummary(summary);
+        break;
+
+      case AsyncOperationType.SOUP_PROMOTE:
+        summary = await this.processSoupDeploy(
+          branch,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          queryResp.sf_devops__Change_Bundle_Installs__r!.records,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          queryResp.sf_devops__Work_Items__r!.records
+        );
+        this.printVersionedOrSoupDeploySummary(summary);
+        break;
+
+      default:
+        // Error here
+        break;
+    }
+  }
+
+  /**
+   * Prints an adHoc deploy summary
+   */
+  private printAdHocDeploySummary(summary: DeploySummary): void {
+    const itemsLabel =
+      (summary.workItems as string[]).length === 1
+        ? output.getMessage('output.item')
+        : output.getMessage('output.items');
+    const workItems = (summary.workItems as string[]).join(', ');
+    console.log(
+      output.getMessage('output.adhoc-deploy-summary', [
+        summary.stageName,
+        itemsLabel,
+        workItems,
+        summary.branchName,
+        summary.orgUrl,
+      ]) + '\n'
+    );
+  }
+
+  /**
+   * Prints a versioned or soup deploy summary
+   */
+  private printVersionedOrSoupDeploySummary(summary: DeploySummary): void {
+    const bundlesSummary: string[] = [];
+    (summary.workItems as Map<string, string[]>).forEach((workItems: string[], bundleVersionName: string) => {
+      const itemsLabel = workItems.length === 1 ? output.getMessage('output.item') : output.getMessage('output.items');
+      bundlesSummary.push(
+        output.getMessage('output.version-or-soup-deploy-summary-wis', [
+          itemsLabel,
+          workItems.join(', '),
+          bundleVersionName,
+        ])
+      );
+    });
+    console.log(
+      output.getMessage('output.version-or-soup-deploy-summary', [
+        summary.stageName,
+        bundlesSummary.join('; '),
+        summary.branchName,
+        summary.orgUrl,
+      ]) + '\n'
+    );
+  }
+
+  /**
+   * Builds a DeploySummary for an adHoc deploy
+   */
+  private async processAdHocDeploy(branchName: string, workItemsPromotes: WorkItemPromote[]): Promise<DeploySummary> {
+    const stageName: string = workItemsPromotes[0].sf_devops__Pipeline_Stage__r.Name;
+
+    const workItems: string[] = workItemsPromotes.map((workItem) => workItem.sf_devops__Work_Item__r.Name);
+
+    const namedCredential: string =
+      workItemsPromotes[0].sf_devops__Pipeline_Stage__r.sf_devops__Environment__r.sf_devops__Named_Credential__c;
+    const orgUrl: string = await selectOrgUrl(this.con, namedCredential);
```

Comment:
As you move to a SummaryProvider, keep in mind _a lot_ of this information is static over the life of the promotion.  We should be able to compute it once.